File: pocketc_backend/src/main/java/com/ssafy/pocketc_backend/domain/transaction/entity/Transaction.java
package com.ssafy.pocketc_backend.domain.transaction.entity;

import com.ssafy.pocketc_backend.domain.user.entity.User;
import com.ssafy.pocketc_backend.global.common.BaseTimeEntity;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "transactions")
public class Transaction extends BaseTimeEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer transactionId;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private LocalDateTime transactedAt;
    private Long amount;

    @Column
    private String merchantName;

    private Integer majorId;
    private Integer subId;

    private boolean isApplied;
    private boolean isFixed;
}

File: pocketc_ai/app/schemas/transaction.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class CategoryRequest(BaseModel):
    amount: int
    merchantName: str
    transactedAt: str

class CategoryResponse(BaseModel):
    transactionId: Optional[int] = None
    amount: int
    merchantName: str
    transactedAt: datetime
    majorId: int
    subId: int
    isFixed: bool         # 고정비 1, 변동비 0

class CategoriesResponse(BaseModel):
    transactions: List[CategoryResponse]
